### PR TITLE
ASoC: SOF: Intel: add SSP support for LunarLake

### PR DESCRIPTION
**Pull request by plbossart**
WIP PR to add nocodec mode support for SSP on LunarLake.

This PR depends on https://github.com/thesofproject/linux/pull/4289, https://github.com/thesofproject/linux/pull/4283 https://github.com/thesofproject/linux/pull/4280 #4303. so will have to be rebased.

### CODE REVIEW

**Comment by greg-intel** [comment]
SOFCI TEST

```
16:50:16  sound/soc/sof/intel/hda-dai.c: In function ‘hda_set_dai_drv_ops’:
16:50:16  sound/soc/sof/intel/hda-dai.c:531:52: error: ‘ssp_dai_ops’ undeclared (first use in this function)
16:50:16    531 |                                 ops->drv[i].ops = &ssp_dai_ops;
16:50:16        |                                                    ^~~~~~~~~~~
16:50:16  sound/soc/sof/intel/hda-dai.c:531:52: note: each undeclared identifier is reported only once for each function it appears in
16:50:16  make[9]: *** [scripts/Makefile.build:252: sound/soc/sof/intel/hda-dai.o] Error 1
16:50:16  make[9]: *** Waiting for unfinished jobs....
16:50:16  make[8]: *** [scripts/Makefile.build:494: sound/soc/sof/intel] Error 2
16:50:16  make[7]: *** [scripts/Makefile.build:494: sound/soc/sof] Error 2
16:50:16  make[6]: *** [scripts/Makefile.build:494: sound/soc] Error 2
16:50:16  make[5]: *** [scripts/Makefile.build:494: sound] Error 2
16:50:16  make[4]: *** [Makefile:2028: .] Error 2
16:50:16  make[3]: *** [Makefile:358: __build_one_by_one] Error 2
16:50:16  make[2]: *** [debian/rules:7: build-arch] Error 2
16:50:16  dpkg-buildpackage: error: debian/rules binary subprocess returned exit status 2
16:50:16  make[1]: *** [scripts/Makefile.package:127: bindeb-pkg] Error 2
16:50:16  make: *** [Makefile:1657: bindeb-pkg] Error 2
```

This is the failure from the Jenkins log, likely due to missing the dependencies mentioned above.

**Comment by marc-hb** [comment]
SOFCI TEST

**Comment by plbossart** [comment]
more work needed, suspend/resume with audio is broken. Gah.

**Review by RanderWang — APPROVED** [review]

File: sound/soc/sof/intel/hda-dai-ops.c [hunk 1]
```diff
@@ -508,6 +513,13 @@ hda_select_dai_widget_ops(struct snd_sof_dev *sdev, struct snd_sof_widget *swidg
 
 			return &hda_ipc4_dma_ops;
 		}
+		case SOF_DAI_INTEL_SSP:
+			if (chip->hw_ip_version < SOF_INTEL_ACE_2_0)
+				return NULL;
```
Comment (ranj063) [inline sound/soc/sof/intel/hda-dai-ops.c:518]: @plbossart will we ever see this in the older platforms?

**Comment by greg-intel** [comment]
SOFCI TEST

Test run almost all green:
https://sof-ci.01.org/linuxpr/PR4290/build4531/devicetest/index.html
https://sof-ci.01.org/linuxpr/PR4290/build4530/devicetest/index.html
https://sof-ci.01.org/linuxpr/PR4290/build4529/devicetest/index.html

One [device failed](https://sof-jenkins.ostc.intel.com/job/testflow/82949/console) with:
`11:39:57 [2023-04-25 18:39:57 UTC][SOF-SH] ERROR: Failed to retrieve Kernel resource from http://sof-storage.ostc.intel.com/pr/linux/pr4290-4528/linux-image-6.3.0-rc3-pr4290-4528-nocodec-g8effe9eab18f_6.3.0-rc3-8effe9ea_amd64.deb`

**Comment by plbossart** [comment]
@ujfalusi @ranj063 can you review, we need to have this in topic/sof-dev

**Review by ranj063 — APPROVED** [review]

**Review by ujfalusi — APPROVED** [review]

**Comment by plbossart** [comment]
> @ujfalusi @ranj063 can you review, we need to have this in topic/sof-dev

thanks for the quick review @ujfalusi @ranj063. Merging now.

**Pull request merged**